diff --git a/default.nix b/default.nix
index 1069e3d..dc4e11f 100644
--- a/default.nix
+++ b/default.nix
@@ -378,13 +378,15 @@ in rec {
             spdir = elemAt (splitString "/" subpath) 0;
           in elem spdir dirsToInclude ||
             (type == "regular" && elem subpath filesToInclude);
-      in builtins.filterSource
-          (mkFilter {
-            dirsToInclude = ["bin" "lib"];
-            filesToInclude = ["package.json" "yarn.lock"];
-            root = src;
-          })
-          src;
+      in if (pkgs.lib.canCleanSource src) then (pkgs.lib.cleanSourceWith {
+            inherit src;
+            filter = (mkFilter {
+              dirsToInclude = ["bin" "lib"];
+              filesToInclude = ["package.json" "yarn.lock"];
+              root = src;
+            });
+          }
+          ) else src;
 
     # yarn2nix is the only package that requires the yarnNix option.
     # All the other projects can auto-generate that file.
