diff --git a/Gopkg.toml b/Gopkg.toml
index a69a858..f237487 100644
--- a/Gopkg.toml
+++ b/Gopkg.toml
@@ -44,14 +44,11 @@ required = [
 
 [[constraint]]
   name = "github.com/knative/eventing-sources"
-  version = "=0.5.0"
+  version = "=0.9.0"
 
-# Constrain the version of knative/pkg we would like to import.
-# This controls when we upgrade apis independently of Serving.
-[[constraint]]
-  name = "github.com/knative/pkg"
-  # HEAD as of 2019-03-21
-  revision = "60fdcbcabd2faeb34328d8b2725dc76c59189453"
+[[override]]
+  name = "knative.dev/pkg"
+  branch = "release-0.9"
 
 [[override]]
   name = "k8s.io/api"
